### PR TITLE
Add more trigonometric functions to backends

### PR DESCRIPTION
**Pull request by brortao**
This PR adds the functions `tan`, `arcsin`, `arccos`, and `arctan` to the Theano and Tensorflow backends. There are tests for these, and I also added tests for the existing functions `sin` and `cos`.

### CODE REVIEW

**Comment by bstriner** [comment]
Please let me know your opinions on an aliasing approach. Keras can provide pass-through, especially for simple functions, and `help(K.asin)` will actually show the documentation for `T.asin`, which is bound to be more detailed and accurate than ours.

Of course this only works for the simplest of functions, but that still means a large number of functions and not bloating the code.

For example, using fancy python assignment:

```
#theano_backend.py
acos, asin, atan = theano.tensor.arccos,  theano.tensor.arcsin,  theano.tensor.arctan
```
```
#tensorflow_backend.py
acos, asin, atan = tf.acos,  tf.asin,  tf.atan
```

Cheers,
Ben

**Comment by brortao** [comment]
Yes, I do think that the aliasing approach is more elegant. If PR https://github.com/fchollet/keras/pull/4932 is merged, I'll close this one!

**Comment by bstriner** [comment]
Seems like vanilla definitions are going to be the easiest for people to understand and maintain, so this PR should be the right way to go. 

Alternatively, it might make more sense to put all of the extra functions into the contrib repo depending on how that pans out. That way you can add something like `K.trig.arctanh` without adding a ton of functions to the backend.

Cheers,
Ben

File: keras/backend/tensorflow_backend.py [hunk 1]
```diff
@@ -1163,6 +1163,30 @@ def cos(x):
     return tf.cos(x)
 
 
+def tan(x):
+    '''Computes tan of x element-wise.
```
Comment (howthebodyworks) [inline keras/backend/tensorflow_backend.py]: Current keras master use double quotes in docstrings AFAICT - `"""`

Comment (brortao) [inline keras/backend/tensorflow_backend.py]: Yes, thank you! Wrt. docstrings: at the time of this writing, `tensorflow_backend.py` has docstrings for simple functions like `tan` while `theano_backend.py` has none. Do you think the goal is to have docstrings for all functions, or is it better if they are skipped for very simple functions?

**Comment by bstriner** [comment]
On the note of documentation, I added this PR https://github.com/fchollet/keras/pull/5123

Documentation is maintained in TF but not theano. This PR copies docstrings from TF backend to theano backend if TF is documented but theano isn't.

**Comment by fchollet** [comment]
Closing outdated PR. If you still care about the content of the PR, please submit a new PR to `master`, updated for the Keras 2.0 API.

**Comment by off99555** [comment]
I need to compute the `arccos` of a number according to https://math.stackexchange.com/a/90098/175303
on how to find angle between quaternions but I don't know how to use `arccos` in `keras.backend`.
How do I do it?

**Comment by off99555** [comment]
OK. I used `tf.acos` to create keras loss function and it looks like it's working.
I need to clip the input to be between -1 and 1 before feeding into `tf.acos` and that's probably fine.
